packages/plugin-ext/src/main/browser/menus/plugin-menu-command-adapter.ts

```diff
@@ -0,0 +1,256 @@
+            ['scm/title', () => this.toScmArg(this.scmService.selectedRepository)],
+            ['timeline/item/context', (...args) => this.toTimelineArgs(...args)],
+            ['view/item/context', (...args) => this.toTreeArgs(...args)],
+            ['view/title', noArgs],
+        ]).forEach(([contributionPoint, adapter]) => {
+            if (adapter) {
+                const paths = codeToTheiaMappings.get(contributionPoint);
+                if (paths) {
+                    paths.forEach(([path]) => this.addArgumentAdapter(path, adapter));
+                }
+            }
+        });
+        this.addArgumentAdapter(TAB_BAR_TOOLBAR_CONTEXT_MENU, widgetURI);
+    }
+
+    canHandle(menuPath: MenuPath, command: string, ...commandArgs: unknown[]): number {
+        if (this.commands.has(command) && this.getArgumentAdapterForMenu(menuPath)) {
+            return 500;
+        }
+        return -1;
+    }
+
+    executeCommand(menuPath: MenuPath, command: string, ...commandArgs: unknown[]): Promise<unknown> {
+        const argumentAdapter = this.getAdapterOrThrow(menuPath);
+        return this.commandRegistry.executeCommand(command, ...argumentAdapter(...commandArgs));
+    }
+
+    isVisible(menuPath: MenuPath, command: string, ...commandArgs: unknown[]): boolean {
+        const argumentAdapter = this.getAdapterOrThrow(menuPath);
+        return this.commandRegistry.isVisible(command, ...argumentAdapter(...commandArgs));
+    }
+
+    isEnabled(menuPath: MenuPath, command: string, ...commandArgs: unknown[]): boolean {
+        const argumentAdapter = this.getAdapterOrThrow(menuPath);
+        return this.commandRegistry.isEnabled(command, ...argumentAdapter(...commandArgs));
+    }
+
+    isToggled(menuPath: MenuPath, command: string, ...commandArgs: unknown[]): boolean {
+        const argumentAdapter = this.getAdapterOrThrow(menuPath);
+        return this.commandRegistry.isToggled(command, ...argumentAdapter(...commandArgs));
+    }
+
+    protected getAdapterOrThrow(menuPath: MenuPath): ArgumentAdapter {
+        const argumentAdapter = this.getArgumentAdapterForMenu(menuPath);
+        if (!argumentAdapter) {
+            throw new Error('PluginMenuCommandAdapter attempted to execute command for unregistered menu: ' + JSON.stringify(menuPath));
+        }
+        return argumentAdapter;
+    }
+
+    addCommand(commandId: string): Disposable {
+        this.commands.add(commandId);
+        return Disposable.create(() => this.commands.delete(commandId));
+    }
+
+    protected getArgumentAdapterForMenu(menuPath: MenuPath): ArgumentAdapter | undefined {
+        return this.argumentAdapters.get(menuPath.join(this.separator));
+    }
+
+    protected addArgumentAdapter(menuPath: MenuPath, adapter: ArgumentAdapter): void {
+        this.argumentAdapters.set(menuPath.join(this.separator), adapter);
+    }
+
+    /* eslint-disable @typescript-eslint/no-explicit-any */
+
+    protected toCommentArgs(...args: any[]): any[] {
```

Comment:
All of these argument converters are old code moved out of the `MenusContributionPointHandler` class, and the code usually has fairly strong assumptions about what the data will look like (not `undefined`, some kind of `object`), so converting the declarations to `unknown` would require a bunch of extra type checks on types where I don't actually really know what's coming in - hopefully the original implementer did. I can go through and do the checks and ensure things have the shape we expect  - I agree that this code is fairly brittle if anything changes - but since it is all copied over, I think I'd prefer to put that in a separate PR.